File: 2ndSemester/kursprogramowania/lista3/cpp/src/pentagon.cpp
#include <pentagon.hpp>

#include <cmath>
#include <string>

Pentagon::Pentagon(double side) noexcept(false)
{
    if (side <= 0)
        throw std::invalid_argument(std::string(__FILE__) + ":" + std::to_string(__LINE__) + " [" + __func__ + "] " + "side length should be positive, got " + std::to_string(side));

    this->side = side;
}

double Pentagon::area() const noexcept(true) 
{
    return 1.7204774005 * this->side * this->side;
}

double Pentagon::perimeter() const noexcept(true)
{
    return 5 * this->side;
}

const std::string Pentagon::name() const noexcept(true) 
{
    return "Pentagon";
}

File: 2ndSemester/kursprogramowania/lista3/java/Pentagon.java
/**
 * Pentagon
 */
public final class Pentagon extends Figure 
{
    private final double side;
    
    public Pentagon(final double side)
    {
        if (side <= 0) 
        {
            throw new IllegalArgumentException("Side length should be positive, got " + side);    
        }
        this.side = side;
    }

    @Override
    public double area() 
    {
        return 1.7204774005 * side * side;
    } 

    @Override
    public double perimeter()
    {
        return 5 * side;
    }

    @Override
    public String name()
    {
        return "Pentagon";
    }
}
